k-distribution/src/main/scripts/lib/k-util.sh

```diff
@@ -11,12 +11,100 @@
 #
 # source "$(dirname "$0")/../lib/kframework/k-util.sh"
 
+time_millis () {
+  # Using this solution to support MacOS
+  # https://superuser.com/questions/599072/how-to-get-bash-execution-time-in-milliseconds-under-mac-os-x
+  # TODO: Is it worth the extra dependency on perl? Should we find a different way.
+  perl -MTime::HiRes -e 'printf("%.0f\n",Time::HiRes::time()*1000)'
```

Comment:
I don't like having to invoke `perl` here, but this was the best way to get milli-second timing information (which is needed for this to be of any use to anyone) in a way supported on both linux and Macos.